### PR TITLE
feat: add map to each result-card

### LINKED ISSUE #20: Show the cordinates coming in the response in a map component,

### PR DESCRIPTION
**Pull request by wise-introvert**
> ### :warning: This feature requires an Access Token provided by the [Mapbox Library](https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/). Please make sure that you add this token to the `.env.` file as `REACT_APP_MAPBOX_ACCESS_TOKEN` ( see `.env.example` )

### Creating the Access Token
- Create a MapBox account by navigating to [this link](https://account.mapbox.com/auth/signup)
image
- After creating and verifying the account, go to https://accoung.mapbox.com and copy the access token
Screenshot from 2021-05-28 13-03-08
- Place the token in `.env` as `REACT_APP_MAPBOX_ACCESS_TOKEN`
---
### Screenshots
image
image
---
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **NEW-UI**. Don't request other protected Branches like **staging/master**
- [x] Your Commit messages should make sense.
- [x] Don't push your package.lock.json as this project uses yarn.lock already.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### Please check if the PR fulfills these requirements

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
  - Feature
- **What is the current behavior?** (You can also link to an open issue here)
  - Results do not have a dedicated map component

- **What is the new behavior (if this is a feature change)?**
  - Each result-card has a map component ( which can be toggled by the user ) that renders the coordinates returned by the API.

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
  - Yes but only in one part of the UI

- **Other Information**:
  - @stephin007 The coordinates returned by the API is wrong and so the resulting render of the map won't match the actual address.
  - Future changes
    - A maker/tooltip component to display the address on the map with the tip pointing to the exact location.
  - Closes #20 

❤️Thank you!

### CODE REVIEW

**Comment by stephin007** [comment]
hey @wise-introvert checking your changes now! it would be great if you could add some screenshots how it looks in the UI.

FYI: As suspected the map wont show correct coordinates as the api is still in draft mode.


As the deployment bot is taking time.

**Comment by stephin007** [comment]
Also @wise-introvert , how do we create the map token: Please describe here 

Thanks

**Comment by stephin007** [comment]
i would need the token as it needs to be added as a secret for the deployment to run automatically.

**Comment by wise-introvert** [comment]
@stephin007 I've added instructions on how to generate the token and screenshots of the updated ui

**Comment by stephin007** [comment]
Alright @wise-introvert thanks, for adding the information, Few more questions:

- When will the map component show? does it open as drop-down when we click on the icon given to the side of the address.
- Can you please add this token generation process in the documentation(i.e. Readme.md) ? in a new PR.
- As an additional feature, will you be able to pinpoint the exact location in the map because that makes the map component complete. 


Please let me know if you have any queries.

**Comment by wise-introvert** [comment]
- The map component is dynamically rendered inside the `<Paper>...</Paper>` component in `VaccineDataSingle.js`. The map icon triggers the render
- Okay, I'll add the process the README
- I am working on the "pinpoint" feature but it may take me some time. We can use the current behavior as MVP though

**Comment by stephin007** [comment]
Got it, @wise-introvert We can use this as a MVP, but can you create another issue for the pinpoint feature for the map as soon as this PR gets merged, so that we can track its progress seperately!

Let me know your thoughts?

**Comment by wise-introvert** [comment]
@stephin007 that sounds good. 

Regarding the documentation: because the process is for contributors only, can we add it as part of issue #53?

**Comment by stephin007** [comment]
Makes sense! Please do @wise-introvert 🙏🏾

**Comment by stephin007** [comment]
hey , @wise-introvert our deployment bot saw that you have an unused variable called "setZoom" Can you please reomve that at commit that in this same PR!
So that we can deploy your branch and see your changes before merging your branch.


image

**Comment by wise-introvert** [comment]
Sure thing

**Comment by wise-introvert** [comment]
@stephin007 have you added the access token to vercel?

**Comment by stephin007** [comment]
yes @wise-introvert , its successfully deployed now! you can view your branch changes in this demo link:  https://cowin-vaccine-availablity-checker-33qvm363l-stephin007.vercel.app/vaccines

Will merge your changes soon

Awesome Job Man 🤝

**Comment by stephin007** [comment]
Checking your commits now @wise-introvert !

File: src/components/VaccineData/Map/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+mapboxgl.accessToken = process.env.REACT_APP_MAPBOX_ACCESS_TOKEN;
+
+export const Map = ({ lng: _lng, lat: _lat, close }) => {
+  const mapContainer = React.useRef(null);
+  const map = React.useRef(null);
+  const [lng, setLng] = React.useState(_lng);
+  const [lat, setLat] = React.useState(_lat);
+  const [zoom] = React.useState(10);
+
+  React.useEffect(() => {
```
Comment (stephin007) [inline src/components/VaccineData/Map/index.js:16]: isn't importing useEffect and useState more reasonable in the first line itself?
we dont have to always do React.useEffect!

Please change these.

File: src/components/VaccineData/SingleVaccineData/VaccineDataSingle.js [hunk 2]
```diff
@@ -12,11 +17,17 @@ const useStyles = makeStyles((theme) => ({
 }));
 const VaccineDataSingle = (vaccine) => {
   const classes = useStyles();
+  const [showMap, setShowMap] = React.useState(false);
+
```
Comment (stephin007) [inline src/components/VaccineData/SingleVaccineData/VaccineDataSingle.js:21]: Here also same issue!

File: src/components/VaccineData/SingleVaccineData/VaccineDataSingle.js [hunk 3]
```diff
@@ -12,11 +17,17 @@ const useStyles = makeStyles((theme) => ({
...
+  React.useEffect(() => {
```
Comment (stephin007) [inline src/components/VaccineData/SingleVaccineData/VaccineDataSingle.js]: here also 😅

**Comment by stephin007** [comment]
Hey @wise-introvert left some comments, please do the changes

**Review by stephin007 — APPROVED** [review]
Looks good to me! 👏🏾

**Comment by stephin007** [comment]
hey @wise-introvert  merging your changes, will soon push this to production as well.

**Pull request merged**